subset/security/brute_test/src/main/java/SshSocket.java

```diff
@@ -0,0 +1,138 @@
+      formattedMac = macAddress.substring(0, 6).toUpperCase();
+      System.out.println("MAC ADDRESS : " + macAddress + "  " + macDevices.get(formattedMac));
+      getJsonFile(formattedMac);
+    } catch (Exception e) {
+      System.out.println(e);
+      Report reportHandler = new Report();
+      reportHandler.addText("RESULT skip security.passwords");
+      reportHandler.writeReport();
+    }
+  }
+
+  public void getJsonFile(String model) {
+    try {
+      JsonObject jsonObject = gson.fromJson(new InputStreamReader(jsonStream), JsonObject.class);
+      JsonObject parent = jsonObject.getAsJsonObject(model);
+      String usernames = parent.get("Usernames").getAsString();
+      String passwords = parent.get("Passwords").getAsString();
+      jsonUsernames = usernames.split(",");
+      jsonPasswords = passwords.split(",");
+      macRetrieved = true;
+    } catch (JsonSyntaxException e) {
+      e.printStackTrace();
+    } catch (JsonIOException e) {
+      e.printStackTrace();
+    } catch (NullPointerException e) {
+      reportHandler = new Report();
+      reportHandler.addText(macAddress);
+      reportHandler.addText("RESULT skip security.passwords");
+      reportHandler.writeReport();
+    }
+  }
+
+  public void connectSshSocket() {
+    while (!testFinished) {
+      if (passwordIndex == jsonPasswords.length) {
+        usernameIndex++;
+        passwordIndex = 0;
+      }
+      if (usernameIndex > jsonUsernames.length - 1) {
+        testFinished = true;
+        reportHandler.addText("RESULT pass " + testName);
+      } else {
+        attempts++;
+        System.out.println(
+            "attempt "
+                + (attempts + 1)
+                + " made using: "
+                + jsonUsernames[usernameIndex]
+                + "/"
+                + jsonPasswords[passwordIndex]);
+        try {
+
+          session = jsch.getSession(jsonUsernames[usernameIndex], host, connectionPort);
+
+          session.setPassword(jsonPasswords[passwordIndex]);
+          try {
+            java.util.Properties config = new java.util.Properties();
+            config.put("StrictHostKeyChecking", "no");
+            session.setConfig(config);
+            session.connect();
+            reportHandler.addText("RESULT fail " + testName);
+            testFinished = true;
+          } catch (JSchException e) {
+            if (e.toString().contains("Connection refused")) {
+              reportHandler.addText("RESULT skip " + testName);
+              System.out.println("SSH not set up on device");
+              testFinished = true;
+              break;
+            } else {
+              passwordIndex++;
+            }
+          }
+        } catch (JSchException e) {
+          System.out.println("session issue");
+          System.out.println(e);
+        }
+      }
+    }
+
+    reportHandler.writeReport();
+    session.disconnect();
+  }
+
+  @Override
+  public void run() {
+    getMACAddress();
+    if (macRetrieved) {

```

Comment:
If the mac address isn't retreived we fail silently?

Reply:
macRetrieved should be renamed to something along the lines "default login credentials received", in the functions on lines 45, and 59 it will skip if it can't get either the mac address or the credentials for that device, and that is done through the try/catch statements

Reply:
Turns out the test still carried on and skipped but not for a lack of valid manufacturer, code has been reformatted to catch this and skip the test